libmamba/src/api/c_api.cpp

```diff
@@ -16,12 +16,26 @@
 #include "mamba/api/list.hpp"
 #include "mamba/api/remove.hpp"
 #include "mamba/api/update.hpp"
+#include "mamba/core/context.hpp"
 
 
+mamba::Context*
+mamba_new_context()
+{
+    return new mamba::Context;
+}
+
+void
+mamba_delete_context(mamba::Context* context)
+{
+    delete context;
+}
+
```

Comment:
Why are we allocating pointers here? Maybe I missed something but I didn't find any call of these allocaters/deleters. Are we giving the responsibility of deleting to the user or is it controlled somehow? Why aren't we using smart pointers instead?

Reply:
This is the C API so there is no smart pointers, only pointers 🥲 
Note that I dont know if the C api is used, but I needed to reflect  the change to make Context usable there.
If we find that we dont need to provide this API, I vote for deleting `c_api.hpp` and `c_api.cpp`

Reply:
Ohh I didn't realize there is one 😮
Yeah I also vote for removing these... ^^

Reply:
The C API is used by Rhumba, so it cannof be deleted yet. If we want to get rid of it, we must provide R bindings through RCPP, but that's mean an additional library.

Reply:
Welp... ok I'll make the necessary fixes.